pkg/service/types.go

```diff
@@ -36,10 +36,16 @@ type MeshService struct {
 	// The name of the service
 	Name string
 
+	// The ClusterDomain of the service
 	ClusterDomain string
 }
 
 func (ms MeshService) String() string {
+	return fmt.Sprintf("%s%s%s%s%s", ms.Namespace, namespaceNameSeparator, ms.Name, namespaceNameSeparator, ms.ClusterDomain)
+}
+
+// NameWithoutCluster returns a string with out the ClusterDomain
+func (ms MeshService) NameWithoutCluster() string {
```

Comment:
```suggestion
func (ms MeshService) Name() string {
```

This would only return the service name

Reply:
let's also add a unit test for this

Reply:
We're changing the service name with this feature. So the new .String(), is closer to the name. There's only 2 places now where we want this format:

1. In the informers, to reference the underlying K8s object.
2. In the secrets, since a Certificate is currently scoped to the global service, but in the future we may want to include this here.

Everywhere else we are changing the name (including how the cluster/services are named)